src/utils/format.ts

```diff
@@ -63,3 +63,16 @@ export const formatNativeTokenValue = (value: bigint, maxDecimalDigits = 6) => {
 export const formatUSDcTokenValue = (value: bigint, maxDecimalDigits = 2) => {
   return formatTokenUnits(value, USDC_DECIMALS, maxDecimalDigits)
 }
+
+export const formatRoundedUSDcTokenValue = (
+  value: bigint,
+  maxDecimalDigits = 2,
+) => {
+  if (value < 10 ** USDC_DECIMALS - maxDecimalDigits)
+    return formatTokenUnits(value, USDC_DECIMALS, USDC_DECIMALS)
+  return formatTokenUnits(value, USDC_DECIMALS, maxDecimalDigits)
```

Comment:
are you sure this code is correct? looks weird and not super clear
e.g. 10 ** 6 - 2 = 999998, not sure you are comparing correctly, maybe parenthesis are needed or something. Would appreciate a variable with a name that explains what's happening

also you could have a variable for last arg with ternary and return function call cause first two args are the same. Something like this
```
const a = condition ? USDC_DECIMALS : maxDecimalDigits;
return formatTokenUnits(value, USDC_DECIMALS, a)
```

Reply:
oh, yes. it should be `USDC_DECIMALS - maxDecimalDigits`

Reply:
simplified that, thank you!